provider/source/data/tzdb/africa

```diff
@@ -0,0 +1,1459 @@
+# tzdb data for Africa and environs
```

Comment:
Question: why did you add this file directly? We can load the text files from a bunch of different places. In the repo, they should live next to the debug source CLDR data.

Reply:
I'm planning to ship these. We're also shipping segmentation data.

Reply:
We only ship the segmentation data because Makoto hand-wrote the rules; the long-term plan is to upstream the hand-written rules into CLDR/Unicode.

I'm surprised that you are the one suggesting that we ship these time zone files in datagen.

Reply:
I'm leaning towards shipping as I want to hide this from clients. I don't want to give the impression that ICU4X data has to be regenerated with each tzdb update.

Reply:
I'm not opposed to that way of thinking, but then we maybe can use `jiff-tzdb`?

Reply:
Another possible option might be to lazy download the files from one of the couple URLs they're available from (thinking about the GitHub mirror or IANA's data URL), and maybe add a TZDB CLI argument to provide either a directory of zoneinfo files to be compiled or a directory of precompiled TZifs.

Reply:
We default to `new_latest_tested` for CLDR data. Seems like we can download clean TZDB files in the same way? We would not pull from main unless the user specifically requests that.

Reply:
> I'm not opposed to that way of thinking, but then we maybe can use jiff-tzdb?

`jiff-tzdb` is a set of embedded zic files, it's not a parser.

I can make it auto-download.

Reply:
Circling back on this, the text sources are actually more powerful than the compiled tzifs. Afaict the tzifs contain only transitions defined in unix timestamps, whereas the source files contain more structure, they call it zoneinfo (bascially what the standard time is for a period), and rules (when dst is observed, and what the offset is). The tzif transitions alone are not sufficient to compute our offset periods.